activerecord/test/cases/associations/eager_load_nested_include_test.rb

```diff
@@ -107,11 +107,11 @@ def setup
     @first_categorization = @davey_mcdave.categorizations.create(category: Category.first, post: @first_post)
   end
 
-  teardown do
-    @davey_mcdave.destroy
-    @first_post.destroy
+  def teardown
```

Comment:
Changing destroying order is for preventing counter_cache updates destroyed inverse_of record (destroying `@first_comment` causes updating destroyed `@first_post`).

https://github.com/rails/rails/blob/v5.1.0.rc2/activerecord/test/models/comment.rb#L9
https://github.com/rails/rails/blob/v5.1.0.rc2/activerecord/lib/active_record/associations/belongs_to_association.rb#L50

Changing `teardown do` to `def teardown` is for easing to see stacktrace.

Before:

```
  1) Error:
EagerLoadNestedIncludeWithMissingDataTest#test_missing_data_in_a_nested_include_should_not_cause_errors_when_constructing_objects:
RuntimeError: can't modify frozen Hash
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/attribute_set.rb:54:in `write_from_user'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/attribute_methods/write.rb:56:in `write_attribute_with_type_cast'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/attribute_methods/write.rb:38:in `write_attribute'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/attribute_methods.rb:359:in `[]='
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/persistence.rb:347:in `increment'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/persistence.rb:358:in `increment!'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/associations/belongs_to_association.rb:50:in `update_counters'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/associations/belongs_to_association.rb:38:in `decrement_counters'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:198:in `block in destroy_row'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:209:in `block in each_counter_cached_associations'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:208:in `each'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:208:in `each_counter_cached_associations'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:194:in `destroy_row'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/locking/optimistic.rb:117:in `destroy_row'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/persistence.rb:195:in `destroy'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/callbacks.rb:321:in `block in destroy'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:97:in `run_callbacks'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:824:in `_run_destroy_callbacks'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/callbacks.rb:321:in `destroy'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:303:in `block in destroy'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:384:in `block in with_transaction_returning_status'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:225:in `block in transaction'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/transaction.rb:194:in `block in within_new_transaction'
    /Users/kamipo/.rbenv/versions/2.4.0/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/transaction.rb:191:in `within_new_transaction'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:225:in `transaction'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:210:in `transaction'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:381:in `with_transaction_returning_status'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:303:in `destroy'
    test/cases/associations/eager_load_nested_include_test.rb:113:in `block in <class:EagerLoadNestedIncludeWithMissingDataTest>'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:413:in `instance_exec'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:413:in `block in make_lambda'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:270:in `block in simple'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:511:in `block in invoke_after'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:511:in `each'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:511:in `invoke_after'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:132:in `run_callbacks'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/testing/setup_and_teardown.rb:45:in `after_teardown'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/fixtures.rb:875:in `after_teardown'
```

After:

```
  1) Error:
EagerLoadNestedIncludeWithMissingDataTest#test_missing_data_in_a_nested_include_should_not_cause_errors_when_constructing_objects:
RuntimeError: can't modify frozen Hash
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/attribute_set.rb:54:in `write_from_user'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/attribute_methods/write.rb:56:in `write_attribute_with_type_cast'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/attribute_methods/write.rb:38:in `write_attribute'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/attribute_methods.rb:359:in `[]='
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/persistence.rb:347:in `increment'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/persistence.rb:358:in `increment!'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/associations/belongs_to_association.rb:50:in `update_counters'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/associations/belongs_to_association.rb:38:in `decrement_counters'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:198:in `block in destroy_row'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:209:in `block in each_counter_cached_associations'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:208:in `each'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:208:in `each_counter_cached_associations'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/counter_cache.rb:194:in `destroy_row'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/locking/optimistic.rb:117:in `destroy_row'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/persistence.rb:195:in `destroy'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/callbacks.rb:321:in `block in destroy'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:97:in `run_callbacks'
    /Users/kamipo/src/github.com/rails/rails/activesupport/lib/active_support/callbacks.rb:824:in `_run_destroy_callbacks'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/callbacks.rb:321:in `destroy'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:303:in `block in destroy'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:384:in `block in with_transaction_returning_status'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:225:in `block in transaction'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/transaction.rb:194:in `block in within_new_transaction'
    /Users/kamipo/.rbenv/versions/2.4.0/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/transaction.rb:191:in `within_new_transaction'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb:225:in `transaction'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:210:in `transaction'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:381:in `with_transaction_returning_status'
    /Users/kamipo/src/github.com/rails/rails/activerecord/lib/active_record/transactions.rb:303:in `destroy'
    test/cases/associations/eager_load_nested_include_test.rb:113:in `teardown'
```